src/comparisons.jl

```diff
@@ -1,7 +1,42 @@
-# This file is inspired from Polyhedra.jl
+# Some functions in this file are inspired from Polyhedra.jl
 
-const ABSZTOL(N::Type{<:AbstractFloat}) = N(10) * sqrt(eps(N))
-const ABSZTOL(N::Type{Rational{INNER}}) where {INNER} = zero(N)
+# struct to contain the tolerances for a given numberic type
+mutable struct TOL{N<:Number}
+    rtol::N
+    ztol::N
+    atol::N
+end
+
+# global Float64 tolerances
+const F64 = Float64
+const tF64 = Type{Float64}
+const _TOL_F64 = TOL(Base.rtoldefault(F64), F64(10)*sqrt(eps(F64)), zero(F64))
+
+get_rtol(N::tF64) = _TOL_F64.rtol
+get_ztol(N::tF64) = _TOL_F64.ztol
+get_atol(N::tF64) = _TOL_F64.atol
```

Comment:
Instead of having a few hardcoded collections like `_TOL_F64`, I suggest to define a dictionary `TOL_N`:
```julia
TOL_N = Dict{Type{<:Number}, Tolerance}()

rtol(N::Type{<:Number}) = get!(TOL_N, N, default_tolerance(N)).rtol
...
xx(N::Type{<:Number}) = get!(TOL_N, N, default_tolerance(N)).xx

set_rtol(N::Type{<:Number}, val) = get!(TOL_N, N, default_tolerance(N))[rtol] = val
...
set_xx(N::Type{<:Number}, val) = get!(TOL_N, N, default_tolerance(N))[xx] = val

default_tolerance(N::Type{Float64}) = Tolerance(..., ..., ...)
...
```

Reply:
Note that your approach only allows to set the bounds for `Float64` and `Rational{Int}`. I find this very restrictive.

Reply:
The idea of a dictionary is appealing, since it allows users to define a new global set of default tolerances without having to manually extend `comparisons.jl` inside LazySets for this purpose (*).

I'm not opposed to using a dictionary whose keys are the numeric types and whose values are instance of `Tolerance`. But i'd like to understand: is there any difference performance-wise between the two approaches? The dictionary cannot be strictly typed (because one cannot know at compile time which numeric types will be added).

(*) of course, `rtol`, `atol` and `ztol` can still be passed as keyword arguments, the feature added in this PR is to set them globally for the numeric type that we use in practice.

Reply:
One can also combine the approaches: use a fixed object for `Float64` and use the dictionary as fallback for all other types.